### ISSUE TITLE
Implement convenient field for first log date

### ISSUE THREAD

**Issue by TomaszGasior** [TomaszGasior/RadioLista-v3#16]
I think about compound field built using three inputs:

SELECT year  | SELECT month | INPUT NUMBER day

Will be this more convenient than current plain text field or not?

**Comment by HyperDX** [comment]
Yes, it will be more comfortable but I have another idea. Choosing a date from the mini pop-up calendar. What do you think?

**Comment by TomaszGasior** [comment]
There are two ways to add this field:
- a) simplier and preferred by me: using built-in calendar provided by popular web browsers (Chrome, Firefox at least)
- b) using external JavaScript fancy calendar. I don't like this solution because it would make RL more heavy.

There is one special requirement for "first log date" field. Normally, when applications have some date-related fields, all information — day AND month AND year — are required from user to be filled in. "first log date" field is different, it **must** allow to user to put date in one of these formats:

- only year OR,
- year AND month,
- year AND month AND day.

Also, application **must** understand difference between `2018`, `2018-01` and `2018-01-01` so we can't assume that first day of month means that we can ignore month at all.

Unfortunately, calendars built in web browsers do not support our requirement of three possible formats — they always require both year and month and day. :( The only possible solution is to use external JavaScript calendar which I personally don't like. :(

Also, there is another question. Are  RadioLista users more keyboard-oriented or more mouse-oriented? If RadioLista users are more keyboard-oriented, current solution based on simple text field is better for them. 

As you can see, it's hard to decide, which direction choose.

**Comment by HyperDX** [comment]
> Also, there is another question. Are RadioLista users more keyboard-oriented or more mouse-oriented? If RadioLista users are more keyboard-oriented, current solution based on simple text field is better for them.

Personally, I prefer typing date using keyboard. For me it's faster than choosing it from calendar ;)

**Comment by TomaszGasior** [comment]
I will keep this issue open but I unassigned myself and added `help wanted` label: maybe someone else would want to implement this feature. Since there are some keyboard oriented users like me or @HyperDX there's one requirement: implementation must offer both ways of entering date: from the keyboard and optionally from visual calendar.

**Comment by TomaszGasior** [comment]
I want to keep here two ideas for implementation.

- Good looking date picker with proper usability: https://github.com/mymth/vanillajs-datepicker
- Input mask for making date format more obvious: https://github.com/RobinHerbots/Inputmask
